lib/y2logsstep.pm

```diff
@@ -207,6 +208,22 @@ sub verify_license_has_to_be_accepted {
     }
 }
 
+sub detect_stuck_at_system_analyzing {
+    my ($self, %args) = @_;
+    my $timeout  = $args{timeout} // 600;
+    my $interval = $args{interval} // 10;
+    my $timetick = 0;
+    my @tags     = qw(analyzing_system adapting_proposal);
+
+    return unless check_screen(\@tags, 10);
+    while (check_screen(\@tags, no_wait => 1)) {
+        sleep 10;
+        $timetick += $interval;
+        last if $timetick >= $timeout;
+    }
+    die "System might be stuck on analyzing" if $timetick >= $timeout;
+}
```

Comment:
Sorry to say that but this looks very unusual. If the test is waiting for release notes why not just use a multi-tag assert screen there and look for 'release-notes-sle' as well as 'analyzing_system'?

Reply:
Yes, it can be done in that way, but similar code will be repeated in 3 different places and looks some bit of duplication, just like:
```
diff --git a/lib/y2logsstep.pm b/lib/y2logsstep.pm
index 2c45ddd3..e33a331d 100644
--- a/lib/y2logsstep.pm
+++ b/lib/y2logsstep.pm
@@ -192,6 +191,18 @@ sub deal_with_dependency_issues {
     if (check_screen 'manual-intervention') {
         $self->deal_with_dependency_issues;
     }
+    # Yast will adapt new settings after confilicts fixed
+    return unless (check_screen 'adapting_proposal');
+    my $timeout  = 600;
+    my $interval = 10;
+    my $timetick = 0;
+
+    while (check_screen('adapting_proposal', no_wait => 1)) {
+        sleep 10;
+        $timetick += $interval;
+        last if $timetick >= $timeout;
+    }
+    die "System might be stuck on adapting" if $timetick >= $timeout;
 }
 
 sub verify_license_has_to_be_accepted {
diff --git a/tests/installation/installation_overview.pm b/tests/installation/installation_overview.pm
index a7c168ad..f5706c47 100644
--- a/tests/installation/installation_overview.pm
+++ b/tests/installation/installation_overview.pm
@@ -24,7 +24,20 @@ sub run {
     record_soft_failure('bsc#1054974') if get_var('ALL_MODULES');
     # overview-generation
     # this is almost impossible to check for real
-    assert_screen "installation-settings-overview-loaded", 150;
+    assert_screen [qw(installation-settings-overview-loaded analyzing_system)];
+    if match_has_tag('analyzing_system') {
+        my $timeout  = 600;
+        my $interval = 10;
+        my $timetick = 0;
+
+        while (check_screen('analyzing_system', no_wait => 1)) {
+            sleep 10;
+            $timetick += $interval;
+            last if $timetick >= $timeout;
+        }
+        die "System might be stuck on analyzing" if $timetick >= $timeout;
+    }
+
 
     $self->deal_with_dependency_issues;
 
diff --git a/tests/installation/releasenotes.pm b/tests/installation/releasenotes.pm
index a86b1f4c..8b76d1ca 100644
--- a/tests/installation/releasenotes.pm
+++ b/tests/installation/releasenotes.pm
@@ -17,7 +17,19 @@ use testapi;
 use version_utils 'is_sle';
 
 sub run {
-    assert_screen('release-notes-button', 60);
+    assert_screen([qw(release-notes-button analyzing_system)], 60);
+    if match_has_tag('analyzing_system') {
+        my $timeout  = 600;
+        my $interval = 10;
+        my $timetick = 0;
+    
+        while (check_screen('analyzing_system', no_wait => 1)) {
+            sleep 10;
+            $timetick += $interval;
+            last if $timetick >= $timeout;
+        }
+        die "System might be stuck on analyzing" if $timetick >= $timeout;
+    }
     return if match_has_tag('bsc#1054478');
```